lib/ansible/galaxy/dependency_resolution/gpg.py

```diff
@@ -0,0 +1,298 @@
+# -*- coding: utf-8 -*-
+# Copyright: (c) 2022, Ansible Project
+# GNU General Public License v3.0+ (see COPYING or https://www.gnu.org/licenses/gpl-3.0.txt)
+"""Signature verification helpers."""
+
+from __future__ import (absolute_import, division, print_function)
+__metaclass__ = type
+
+from ansible.errors import AnsibleError
+from ansible.module_utils.common.text.converters import to_text
+from ansible.module_utils.urls import open_url
+from ansible.utils.display import Display
+
+import os
+import subprocess
+import textwrap
+
+from dataclasses import dataclass, fields as dc_fields
+from urllib.error import HTTPError
+
+display = Display()
+
+
+def run_gpg_verify(b_manifest_file, signature, keyring):
+    status_fd_read, status_fd_write = os.pipe()
+
+    cmd = [
+        'gpg',
+        f'--status-fd={status_fd_write}',
+        '--verify',
+        '--batch',
+        '--no-tty',
+        '--no-default-keyring',
+        f'--keyring={keyring}',
+        '-',
+        b_manifest_file,
+    ]
+
+    try:
+        p = subprocess.Popen(
+            cmd,
+            stdin=subprocess.PIPE,
+            stdout=subprocess.PIPE,
+            stderr=subprocess.PIPE,
+            pass_fds=(status_fd_write,),
+        )
+        stdout, stderr = p.communicate(input=signature)
+    except subprocess.SubprocessError as err:
+        cmd_str = ' '.join([to_text(arg) for arg in cmd])
```

Comment:
No need to create an in-memory list:
```suggestion
        cmd_str = ' '.join(to_text(arg) for arg in cmd)
```
Also, maybe consider constructing this and logging with info level before "try".

Reply:
Fixed, and added a display.vvvv message with the command (and result at the end of the function) for debugging purposes.